### PR TITLE
scripts/checkdeps improvements

### PR DESCRIPTION
**Pull request by stefansaraev**
as commit msgs say.

### CODE REVIEW

**Comment by stefansaraev** [comment]
@lrusak does fedora/centos/rhel commit look ok ?

**Comment by sraue** [comment]
from first look without testing anything it looks good so far, i would make "scripts/checkdeps: clean up" even more simple with moving the "echo's and read's" before the distrocheck?

**Comment by stefansaraev** [comment]
I'll squash all later.

**Comment by stefansaraev** [comment]
added one more clean up commit. there is no need for checkdeps get / qemu.

File: scripts/checkdeps [hunk 1]
```diff
@@ -75,64 +69,35 @@ done
...
-    if ([ "$ans" = "y" ] && ! [ $(sudo -n uptime 2>&1|grep "load"|wc -l) ]); then
-      sudo aptitude install $need_pkg
-    elif [ "$ans" = "y" ]; then
-      su -c "aptitude install $need_pkg"
-    else
-      echo "********"
+      ;;
+    fedora|centos|rhel)
+      read -p "would you like to install the needed tools ? (y/n) " ans
+      [ "$ans" = "y" ] && sudo yum install $need_pkg
```
Comment (lrusak) [inline scripts/checkdeps]: Might need to check for yum as I don't have it on my machine, dnf is the replacement on newer versions

File: scripts/checkdeps [hunk 2]
```diff
@@ -20,32 +20,26 @@
...
+fi
+
+deps="wget bash bc gcc g++ sed patch touch tar bzip2 gzip perl cp gawk makeinfo gperf zip unzip mkfontscale mkfontdir bdftopcf diff xsltproc java md5deep"
+deps_pkg="wget bash bc gcc g++ sed patch fileutils tar bzip2 gzip perl coreutils gawk texinfo gperf zip unzip xfonts-utils xfonts-utils xfonts-utils diff xsltproc default-jre md5deep"
+files="/usr/include/stdio.h /usr/include/ncurses.h"
+files_pkg="libc6-dev libncurses5-dev"
+
+case "$DISTRO" in
+    fedora|centos|rhel)
+      deps_pkg="wget bash bc gcc gcc-c++ sed patch fileutils tar bzip2 gzip perl coreutils gawk texinfo gperf zip unzip xorg-x11-font-utils xorg-x11-font-utils diffutils libxslt java-1.7.0-openjdk"
```
Comment (lrusak) [inline scripts/checkdeps]: We talked about this before I think and we need glibc-static and libstdc++-static on fedora

Comment (stefansaraev) [inline scripts/checkdeps]: sadly. this can't be done. not in deps. fedora is a bit fucked up.

File: scripts/checkdeps (hunk 1 above)
Comment (stefansaraev) [inline scripts/checkdeps]: unrelated to this pr :)

File: scripts/checkdeps (hunk 2 above)
Comment (lrusak) [inline scripts/checkdeps]: why not? this is need for centos as well I'm sure.

Comment (sraue) [inline scripts/checkdeps]: this cant be done easy with this script, this would need a complete rework (which we should do later)

Comment (lrusak) [inline scripts/checkdeps]: May I'm just not really understanding the script, but can't you just do
```
case "$DISTRO" in
    fedora|centos|rhel)
        deps="$deps glibc-static libstdc++-static"
        deps_pkg="wget bash bc gcc gcc-c++ sed patch fileutils tar bzip2 gzip perl coreutils gawk texinfo gperf zip unzip xorg-x11-font-utils xorg-x11-font-utils diffutils libxslt java-1.7.0-openjdk glibc-static libstdc++-static"
```

Comment (stefansaraev) [inline scripts/checkdeps]: nope. and again. this is no related to this PR. this pr is just clean up / minor enhancements (and a fix for my debian jessie host). no behaviour changed for fedora..

**Comment by lrusak** [comment]
:+1: push the button

**Comment by stefansaraev** [comment]
ready to go. thanks for fixing fedora & arch & gentoo @lrusak

**Pull request closed without merging**